partner_second_lastname/models/res_partner.py

```diff
@@ -61,24 +69,28 @@ def _get_inverse_name(self, name, is_company=False):
         - Otherwise, make a guess.
         """
         # Company name goes to the lastname
-        if is_company or not name:
-            parts = [False, name or False, False]
-
-        # The comma separates the firstname
-        elif "," in name:
-            lastnames, firstname = name.split(",", 1)
-            parts = [firstname.strip()] + lastnames.split(" ", 1)
-
-        # Without comma, the user wrote the firstname first
-        else:
-            parts = name.split(" ", 2)
-
-        while len(parts) < 3:
-            parts.append(False)
-
-        return {"firstname": parts[0],
-                "lastname": parts[1],
-                "lastname2": parts[2]}
+        result = {
+            'firstname': False,
+            'lastname': name or False,
+            'lastname2': False,
+        }
+        if not is_company and name:
+            order = self._get_names_order()
+            result = super(ResPartner, self)._get_inverse_name(
+                name, is_company)
+            if order == 'last_first':
+                parts = result['firstname'].split(" ", 1)
+                while len(parts) < 2:
+                    parts.append(False)
+                result['lastname2'] = parts[0]
+                result['firstname'] = parts[1]
```

Comment:
If ```order == 'last_first'``` then ```lastname``` is only one word, but in other cases ```firstname``` is one word. This is how partner_firstname addon works.